web_extension/shared/options.js

```diff
@@ -101,3 +102,55 @@ class PhotoHistory extends OptionsSubset {
     }
   }
 }
+
+class PhotoCache {
+  constructor () {
+    this.items = []
+    this.last_new_image = 0 // "early" epoch timestamp
+    this.fetched = false
+  }
+
+  async read () {
+    const stored_options = await browser.storage.local.get()
+    this.items = stored_options.items
+
+    if (! Array.isArray(this.items))
+      this.items = []
+
+    if (stored_options.last_new_image)
+      this.last_new_image = stored_options.last_new_image
+
+    this.fetched = true
+  }
+
+  async write () {
+    return browser.storage.local.set(this)
```

Comment:
do you need to `await` this maybe because isn't this a promise that resolves to a promise?

Reply:
I don't need to wait on this anywhere, so I didn't